lib/src/main/kotlin/com/swmansion/starknet/extensions/ToParam.kt

```diff
@@ -0,0 +1,10 @@
+package com.swmansion.starknet.extensions
+
+import com.swmansion.starknet.data.types.BlockId
+
+@JvmSynthetic
+internal fun BlockId.toParam(): Pair<String, String> = when (this) {
```

Comment:
This is only used in GatewayProvider, why not define it there as a helper function?

Reply:
It would complicate testing of this function. While generally we probably shouldn't need to tests internals like this, issue this PR is resolving is caused by us not covering these cases (because test e2e test are incomplete and because of devnet limitations).